### ISSUE TITLE
Maintenance 2024-10

### ISSUE THREAD

**Issue by amotl** [kennethreitz/responder#525]
Hi Kenneth. Thanks for [adding me](https://github.com/kennethreitz/responder/issues/522#issuecomment-2433662104) to the project. Let's use this ticket to enumerate and discuss relevant details about maintenance and friends?

**Comment by amotl** [comment]
@kennethreitz said:

> - getting tests to pass is primary goal, as well as updating starlette and python compatibility.

@amotl suggests:

> - Let's also rename branch master to main right away? Because I don't have ownership permissions, I would need you to conduct this operation.

**Comment by kennethreitz** [comment]
@amotl sure i'll do that now

**Comment by kennethreitz** [comment]
done

**Comment by kennethreitz** [comment]
@coderabbitai what do you think?

**Comment by kennethreitz** [comment]
^^ an experimental bot to help with maintenance

**Comment by kennethreitz** [comment]
@coderabbitai can you highlight the differences between the current starlette version and latest?

**Comment by amotl** [comment]
Thanks! Default branch still shows as `master` for me.

![image](https://github.com/user-attachments/assets/e9a09fcd-86a1-4d65-bd9b-42d83132635a)

**Comment by kennethreitz** [comment]
main is now live

**Comment by kennethreitz** [comment]
@amotl thanks so much for helping!!

**Comment by amotl** [comment]
Do you want to review such chore PRs like GH-526 and GH-527, or should I just go ahead and merge them as long as they don't break CI?

**Comment by kennethreitz** [comment]
@amotl go ahead and merge. I'm getting my GitHub notifications fixed still. They all go to an 'unsubscribed' folder in my inbox which is super annoying.

**Comment by amotl** [comment]
In order to support a linear history on this project, can I ask you to configure its settings like that?

![image](https://github.com/user-attachments/assets/9aabf333-7c0b-4e4f-b3d4-3955ee9231c5)

I guess you could also disable Discussions, Projects, and Wiki, in order to trim the tab navigation.

**Comment by kennethreitz** [comment]
yes will do now

**Comment by kennethreitz** [comment]
@amotl done

**Comment by kennethreitz** [comment]
@amotl username on pypi?

**Comment by amotl** [comment]
On PyPI, I am also amotl.

**Comment by amotl** [comment]
Another humble request: Please don't use squash merging by default, specifically on a bundled PR where individual commits are, well, individual changes, often unrelated.

![image](https://github.com/user-attachments/assets/d905e8e7-4157-4258-9f3d-85abe63dde1f)

**Comment by amotl** [comment]
Can I humbly ask you to have a closer look at those patches?

- GH-531
- GH-532

Afterwards, I think we could think about publishing a release, probably v2.1.0? Will you run the release procedure, or will you add me on PyPI, so I can do it?

**Comment by amotl** [comment]
Thanks a stack. Those are next:

- GH-532
- GH-543

**Comment by kennethreitz** [comment]
@amotl you now have publishing permissions on PyPi.

**Comment by amotl** [comment]
Hi again. Thanks for merging all my patches. GH-535 will take more time, so I guess we are now ready to release version 2.1.0?

**Comment by amotl** [comment]
When you want me to run the release, can you please provide a few guidelines how you would like the CHANGELOG.md to be updated?

**Comment by amotl** [comment]
Hi again. Another round of patches is ready for review and integration.

- GH-549
- GH-554
- GH-555
- GH-556
- GH-552

When merging individual PRs, others may need to be refreshed, including resolving conflicts. Just ping me about it.

NB: With GH-549, I detoured a bit into bot-assisted yak shaving, addressing many suggestions by @coderabbitai. It was an interesting and valuable conversation. Currently, I can't provide the same efforts for GH-554, as it easily takes multiple hours to do the bot dance well.

**Comment by amotl** [comment]
Hi Kenneth,

in order to move forward, I guess applying an ack-by-timeout procedure on reviewing PRs will also be fine, given that you merged all of my previous PRs without much ado, and pretending that we have a shared mindset to produce a good outcome for the community of Responder users, both in terms of project maintenance, and gradually improving its features on a few minor details.

Please don't hesitate to have your voice also on already-merged-PRs. Because "it's just software", it is reasonably easy to roll back individual changes when there are any concerns about them, even in retrospective. @tabotkevin is also occasionally having an eye, so I guess it will all be just fine to move forward.

I hope you agree with this procedure, otherwise please let me know.

With kind regards,
Andreas.

**Comment by kennethreitz** [comment]
Yes, I think that's a good plan. Sorry that I've been a little delayed on contributing. I like all the work you've been doing, but have been focused on other things.

My only request is keeping alabaster and being cautious with switching to markdown.

**Comment by kennethreitz** [comment]
I trust you

**Comment by kennethreitz** [comment]
@amotl do you want me to disable coderabbit on this repo?

**Comment by kennethreitz** [comment]
Sounds great :)Sent from my iPhoneOn Oct 29, 2024, at 6:20 PM, Andreas Motl ***@***.***> wrote:﻿
The bot is sometimes is a bit on the edge of spamming, definitively quite verbose, but it has good intentions, and indeed it was tremendously helpful and supportive on some occasions. For now, unless there are any other objections about it, why not keep it for the time being?

**Comment by amotl** [comment]
About leveraging LLM bots for code reviews, this one just appeared on my radar through a post on HN.

- https://github.com/marketplace/llamapreview
- https://news.ycombinator.com/item?id=41996859

I am sure there will be more of that in the future. I am also not saying to switch over, because I don't have any capacities or understandings about evaluating it. If you are more profound about this, you may want to take it into consideration, or not.

**Comment by kennethreitz** [comment]
@amotl I enabled it on this repo! Let's see if it's any good.

**Comment by kennethreitz** [comment]
Hopefully the bots don't start fighting hahaha

**Comment by amotl** [comment]
Omg. I am sure they will.

**Comment by amotl** [comment]
Back to serious work: Why don't the bots automatically perform those kinds of chores for us? ;] 🍀 
Until they will do it, it is probably still our job. Please don't forget about enabling the project on RTD:

- https://github.com/kennethreitz/responder/issues/564#issuecomment-2447859150

**Comment by adrian-kuhn** [comment]
Hi Andreas, hi Kenneth
First of all, I love the _responder_ library and I use it in several projects. It's so handy and lightweight - just fantastic. I actively recommend it when someone asks me for an API solution. Many thanks for all of your work with this great peace of software!

I see there's a release on the horizon supporting python 3.10+. Can you make a prediction when this release will be published? We could then align the release planning and the migration to python 3.10+ of our products.

Thank you for your answer,
Adrian

**Comment by adrian-kuhn** [comment]
Hi Andreas, thank you for your answer. Looking forward to it!

Let me know, if I can support you somehow. Unfortunately, I don't have any experience with library implementations and Github colaboration and contribute processes. I would have to learn it first.

**Comment by amotl** [comment]
Hi again. [responder==3.0.0.dev0](https://pypi.org/project/responder/3.0.0.dev0/) has been released, and we will be happy to hear back from you testing it. We may close this ticket, so please optimally respond over here:

- GH-578

**Comment by amotl** [comment]
Hi again. I think it will be safe to close this ticket, as a lot of maintenance cycles have been concluded well, in 2024 already. Please follow up on the discussion by having your say on converging the current status quo into the upcoming release.

- GH-578

Thank you very much for your support. ✨
